Kickstarter-iOS/Views/Controllers/DiscoveryPageViewController.swift

```diff
@@ -240,16 +240,16 @@ internal final class DiscoveryPageViewController: UITableViewController {
     self.navigationController?.pushViewController(vc, animated: true)
   }
 
-  fileprivate func showEmptyState(_ emptyState: EmptyState) {
-    guard let emptyVC = self.emptyStatesController else { return }
+  fileprivate func showEmptyState(_ emptyState: EmptyState?) {
```

Comment:
Why is `EmptyState` optional now?

Reply:
One of the problems that was causing the bug is that the `showEmptyState` signal was emitting when it shouldn't (Like for example emitting empty array of projects while the projects were not done loading). So when it emits in any situation where is doesn't match the requirement (projectsAreLoading == false AND projects.isEmpty), it returns nil. So the ViewController receives a nil value, it won't show the EmptyState view.

Reply:
Actually your suggestion of using filter after combineLatest was a good one and EmptyState is no longer optional.